cmd/disk-cache.go

```diff
@@ -270,8 +270,10 @@ func (c *cacheObjects) GetObjectNInfo(ctx context.Context, bucket, object string
 			cacheReader.Close()
 			c.cacheStats.incMiss()
 			bReader, err := c.InnerGetObjectNInfoFn(ctx, bucket, object, rs, h, opts)
-			bReader.ObjInfo.CacheLookupStatus = CacheHit
-			bReader.ObjInfo.CacheStatus = CacheMiss
+			if bReader != nil {
```

Comment:
Can we do something like below? In case of error anyway I feel `bReader` should not be usable from caller
```
if err != nil {
    return nil, err
}
bReader.ObjInfo.CacheLookupStatus = CacheHit
bReader.ObjInfo.CacheStatus = CacheMiss
```

Reply:
Yeah